src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -1919,7 +1919,7 @@ AuthorizedAuthenticationState doAuthorization(final boolean requireLoginPermissi
             final AuthorizationIdentity authorizationIdentity = realmIdentity.getAuthorizationIdentity();
             final SecurityDomain domain = capturedIdentity.getSecurityDomain();
 
-            SecurityIdentity authorizedIdentity = Assert.assertNotNull(domain.transform(new SecurityIdentity(domain, authenticationPrincipal, realmInfo, authorizationIdentity, domain.getCategoryRoleMappers(), capturedIdentity.getPublicCredentials(), capturedIdentity.getPrivateCredentialsPrivate())));
+            SecurityIdentity authorizedIdentity = Assert.assertNotNull(domain.transform(new SecurityIdentity(domain, authenticationPrincipal, realmInfo, authorizationIdentity, domain.getCategoryRoleMappers(), IdentityCredentials.NONE, IdentityCredentials.NONE)));
```

Comment:
I think this will break PLAIN credential forwarding: the PLAIN server propagates the credentials before authorization.  Other mechanisms might do the same.